packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -87,6 +95,25 @@ library TaikoData {
         address sender; // a.k.a proposer
     }
 
+    struct BlockMetadata2 {
```

Comment:
We could still use `BlockMetadata` if we want to, or any specific reason for this change? We could just use the values as they were previously (that's what we do on the helder deployment). I guess maybe part of the optimization for the storage writing, but still stored in `Transition` below so not sure.

Reply:
I would like to take this hard fork opportunity to further optimize storage reads/writes. With this v2 metadata definition, I removed writing "proposedAt" and "proposedIn" into the Block struct in storage. The previous two fields inside the Block struct "proposedAt", and "proposedIn" are used as the L2 block's "timestamp" and "anchorBlockHeight" instead. Otherwise, we'll introduce two new fields into the Block struct.